cli/deno_dir.rs

```diff
@@ -32,8 +28,37 @@ use std::sync::Mutex;
 use url;
 use url::Url;
 
-const SUPPORTED_URL_SCHEMES: [&str; 3] = ["http", "https", "file"];
+#[derive(Debug, Clone)]
+pub struct SourceFile {
+  pub url: Url,
+  pub redirect_source_url: Option<Url>,
+  pub filename: PathBuf,
+  pub media_type: msg::MediaType,
+  pub source_code: Vec<u8>,
+}
+
+impl SourceFile {
+  // TODO: this method should be implemented on CompiledSourceFile trait
+  pub fn js_source(&self) -> String {
+    if self.media_type == msg::MediaType::TypeScript {
+      panic!("TypeScript module has no JS source, did you forget to run it through compiler?");
+    }
+
+    // TODO: this should be done by compiler and JS module should be returned
+    if self.media_type == msg::MediaType::Json {
+      return format!(
+        "export default {};",
+        str::from_utf8(&self.source_code).unwrap()
+      );
+    }
+
+    // it's either JS or Unknown media type
+    str::from_utf8(&self.source_code).unwrap().to_string()
+  }
+}
 
+// TODO: this should be removed, but integration test 022_info_flag depends on
+// using "/" (forward slashes) or doesn't match output on Windows
```

Comment:
Ok, let's do this in a follow up commit. We can just extend the wildcard in the first line of `tests/022_info_flag.out`

Reply:
https://github.com/denoland/deno/pull/2660